### DISCUSSION TITLE
Don't issue mode 2031 DSR reports when colors are changed by a VT sequence

### DISCUSSION THREAD

**Discussion by gpanders** [ghostty-org/ghostty#3933] [Feature Requests, Ideas] [answered]
After discussion in the [Contour issue tracker](https://github.com/contour-terminal/contour/issues/1659) and in Ghostty's Discord, @christianparpart suggested that Ghostty should not be sending mode 2031 DSRs when any color changes based on a VT sequence (e.g. an OSC 10/11/12 sequence). DSRs should only be sent when the color palette changes because of a config reload or system event such as a transition from light to dark mode.

For Neovim this would solve the issue of receiving too many DSRs (we currently get one every time the cursor color changes, which can be quite frequent based on the user's `'guicursor'` setting).

**Comment by mitchellh** [comment, marked as answer]
approved #3965
